lib/filesystem.nix

```diff
@@ -42,4 +44,25 @@
               type = (builtins.readDir parent).${base} or null;
           in file == /. || type == "directory";
     in go (if isDir then file else parent);
+
+  # filterMapDir :  ( String -> String -> Bool )
+  #              -> ( Path -> String -> a )
+  #              -> Path
+  #              -> { Path = a }
+  # This maps a function `f` over the entries filtered by `filter` in a directory located at `path`. 
+  # It doesn't recurse over subdirectories.
+  filterMapDir = filter: f: path:
+    #Sets are lexically sorted by key
+    let files = builtins.readDir path;
```

Comment:
Why hard-code the `readDir`?

If you change the signature to take the resulting data structure of `readDir` instead of a `Path`, the user can chain multiple invocations (and do other transformations), and the function becomes more composable.

Reply:
My intention is to have a convenience function that lets you apply a function to a directory without having to do anything else, but sure, I don't see why I couldn't make another utility function that does what you said - and then use that to implement the convenience function.

Reply:
I’d argue it’s still convenient, as long as the usage is in the example.